plugins/exporter.koplugin/target/json.lua

```diff
@@ -16,6 +50,10 @@ local function format(booknotes)
         file = booknotes.file,
         number_of_pages = booknotes.number_of_pages
     }
+    if settings.bookChecksum then
```

Comment:
something like:

```lua
    local t = {
        title = booknotes.title,
        author = booknotes.author,
        entries = {},
        exported = booknotes.exported,
        file = booknotes.file,
        number_of_pages = booknotes.number_of_pages,
        md5sum = settings.bookChecksum and md5.sumFile(booknotes.file),
    }
```

will get the same output

Reply:
this exports a field `md5sum = false` is that fine ?

Reply:
mmm, I guess it happens only when you flip the setting a couple of times.

In any other context I would recommend to use https://github.com/koreader/koreader/blob/c9ff0071e384d3ab7491fc4c4d6da967f4d643de/frontend/luasettings.lua#L156

but here is a bit abstracted away and not easiest thing to do.

You can do something like this on the callback:
```lua
callback = function()
    self.settings.bookChecksum = self.settings.bookChecksum and nil or true
    self:saveSettings()
end,
```

for this PR and we can refactor that bit of code later if some other exporters start to abuse this kind of construct

Reply:
I think you misunderstood my last message, I meant the code you proposed worked perfectly as you said, when it's set to `true` the checksum is exported. But when it's set to `false` there is a useless checksum field in the json is set to false. 

The change however on your last comment does not work it makes the option sticky.

Reply:
lol, sorry. sir Boole would be very angry with me :)

Please do what @poire-z suggested, and sorry for the extra time lost :p

Reply:
@pazos  @poire-z perfect thanks ! Now the field set to false means `nil` and is actually not exported in json. So this option is incognito until the user needs it :)

Reply:
> Please do what @poire-z suggested, and sorry for the extra time lost :p

no pasa nada 😄